### ISSUE TITLE
Crash with gen3D on structures like "c1ccc(F)cc1"

### ISSUE THREAD

**Issue by zuranski** [openbabel/openbabel#2117]
- [x] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version: 3.0.0
Operating system and version: Win 10

## Expected Behavior

Openbabel 2.4.1 generates such structures correctly, 3.0.0 crashes every time.

## Steps to Reproduce
Generate 3D coordinates for a structure "c1ccc(F)cc1" using either the GUI or using python bindings.

**Comment by fredrikw** [comment]
I can reproduce this. If I change F to Br, the crash disappears but the resulting coordinates are bad:
```
>obabel -ismi -:"c1ccc(Br)cc1" -omol --gen3D

 OpenBabel02192013203D

  7  7  0  0  0  0  0  0  0  0999 V2000
    0.0000    0.0000    0.0000 C   0  0  0  0  0  3  0  0  0  0  0  0
    0.0000    0.0000   -0.0000 C   0  0  0  0  0  3  0  0  0  0  0  0
    0.0000   -0.0000    0.0000 C   0  0  0  0  0  3  0  0  0  0  0  0
   -0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -1.5020    1.1970    0.0000 Br  0  0  0  0  0  0  0  0  0  0  0  0
    0.0000   -0.0000    0.0000 C   0  0  0  0  0  3  0  0  0  0  0  0
   -0.0000    0.0000    0.0000 C   0  0  0  0  0  3  0  0  0  0  0  0
  1  2  1  0  0  0  0
  1  7  2  0  0  0  0
  2  3  2  0  0  0  0
  3  4  1  0  0  0  0
  4  5  1  0  0  0  0
  4  6  2  0  0  0  0
  6  7  1  0  0  0  0
M  END
1 molecule converted
```

**Comment by jrhaulung** [comment]
Hi 
   I also encounter similar situation when converting a bunch of  Zinc15-biogenic compounds in smile strings to generate 3D structure using --gen3D to output as mol2 format in my Ubuntu machine (16.04 LTS) using openbabel 3.0.0.  Screen freeze, process killed itself in the middle after freeze and many output with no 3D coordinate like above occur very frequently. Since there is no crash output in /var/crash and error information  in auth.log in /var/log. Is there anything I could do to resolve this question, Your kindly help will be highly appreciated.

sincerely

Jrhau

**Comment by hjjvandam** [comment]
This one seems similar to #2141. See discussion at #2144.

**Comment by ghutchis** [comment]
This is fixed in `master` and v3.1.
